src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -118,6 +119,27 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
       else Iterator.empty.next()
   }
 
+  def takeRight(n: Int): Iterator[A] = {
+    val buffer = new Array[Any](n)
+    var i = 0
+    var m = 0
+    while (n > 0 && self.hasNext) {
+      buffer.update(i, self.next())
+      i = (i + 1) % n
```

Comment:
Ow, modulus every iteration?  A branch is usually considerably faster.  `if (i >= n) 0 else i+1`